### PR TITLE
Fixing accesibility for user and channel search input options

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing accesibility problem on and user autocomplete inputs on invitation modal.

#### Ticket Link
[MM-17538](https://mattermost.atlassian.net/browse/MM-17538)

### CODE REVIEW

**Comment by lindy65** [comment]
Thanks @jespino - please let me know once dev review is done and I'll go ahead and QA. Thanks!

File: components/widgets/inputs/users_emails_input.jsx [hunk 1]
```diff
@@ -94,6 +94,24 @@ export default class UsersEmailsInput extends React.Component {
...
+        let fullName = null;
+        if (parts.fullName) {
+            fullName = parts.fullName;
+        }
+        let nickname = null;
+        if (parts.nickname) {
+            nickname = parts.nickname;
+        }
+
+        return `${parts.displayName || ''} ${fullName || ''} ${nickname || ''}`;
```
Comment (Willyfrog) [inline components/widgets/inputs/users_emails_input.jsx]: I don't think the blank spaces are needed if elements are missing, right?
```suggestion
        const results = [];
        if (parts.displayName) {
             results.push(parts.displayName);
        }
        if (parts.fullName) {
            results.push(parts.fullName);
        }
        if (parts.nickname) {
            results.push(parts.nickname);
        }
    return results.join(' ');
```

Comment (jespino) [reply components/widgets/inputs/users_emails_input.jsx:112]: for the context is not specially important, but I think you are right, is more correct for sure, I'll apply your proposed changes.

**Comment by lindy65** [comment]
Hi @jespino,

There is no issue adding the first person to invite - Jaws announces the email address as I type it. However,

1. If I just click in the "Add or Invite People" field on the Invite Members to [teamname] screen, (as opposed to tabbing through the screen to get to the field), and type the first user's email address, Jaws reads out "Zero results found for search term alice@mattermost.com" even although the user is suggested in the list below.  Strangely enough, if I leave that browser tab and then return to it, Jaws then reads out "1 result found for alice@mattermost.com"

    - 1a) If I navigate to the "Add or Invite People" field using TAB, Jaws still reads out "Zero results found for search term alice@mattermost.com" even though the user is suggested below (as per screenshot).  So it doesn't seem to make any difference whether I just click in the field and start typing or whether I TAB to the field.

    - 1b) Weirdly - I cannot reproduce this every time - sometimes this happens when I enter the first user's email address and at other times, it only happens after I've selected the first user to add and type the second user.

image
 
2. If I just click in the "Add or Invite People" field on the Invite Members to [teamname] screen, (as opposed to tabbing through the screen to get to the field), Jaws _sometimes_ loses focus after selecting the first person I invite and then reads out some weird stuff as per screenshot:

image

Apologies if this is confusing / not well explained - it's difficult to explain Jaws behavior without actually listening/viewing it on the screen to see for oneself what it's doing...

**Comment by esethna** [comment]
@jespino what's next steps on this PR?

**Comment by hahmadia** [comment]
> @jespino what's next steps on this PR?

@esethna 
Abandoning it I guess 🤣

**Comment by jespino** [comment]
@esethna @hahmadia Actually, yes, I think it should be transferred to somebody more used to deal with accessibility.

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.

**Pull request closed without merging**